### PR TITLE
Docs: Create soc.md

### PR DESCRIPTION
**Pull request by dyzheng**

### CODE REVIEW

**Review by caic99 — APPROVED** [review]

File: docs/advanced/scf/soc.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+## SOC Effects
+### SOC 
+`lspinorb` is used for control whether or not SOC(spin-orbit coupling) effects should be considered.
+
+Both `basis_type=pw` and `basis_type=lcao` support `scf` and `nscf` calculation with SOC effects.
+
+Atomic forces and cell stresses can not calculated with SOC effects until now.
```
Comment (caic99) [inline docs/advanced/scf/soc.md]: In my humble opinion, "until now" means you can not do something before, but you can do it from now on. I guess this is not what you mean.
```suggestion
Atomic forces and cell stresses can not be calculated with SOC effects yet. 
```

File: docs/advanced/scf/soc.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+## SOC Effects
+### SOC 
+`lspinorb` is used for control whether or not SOC(spin-orbit coupling) effects should be considered.
+
+Both `basis_type=pw` and `basis_type=lcao` support `scf` and `nscf` calculation with SOC effects.
+
+Atomic forces and cell stresses can not calculated with SOC effects until now. 
+
+### Pseudopotentials and Numerical Atomic Orbitals
+For Norm-Conserving pseudopotentials, there are difference between SOC version and non-SOC version.
```
Comment (caic99) [inline docs/advanced/scf/soc.md]:
```suggestion
For Norm-Conserving pseudopotentials, there are differences between SOC version and non-SOC version.
```

File: docs/advanced/scf/soc.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+Please check your pseudopotential files before calculating.
+In `PP_HEADER` part, keyword `has_so=1` and `relativistic="full"` refer to SOC effects have been considered, 
+which would lead to different `PP_NONLOCAL` and `PP_PSWFC` parts.
+Please be careful that `relativistic="full"` version can be used for SOC or non-SOC calculation, but `relativistic="scalar"` version only can be used for non-SOC calculation
```
Comment (caic99) [inline docs/advanced/scf/soc.md]:
```suggestion
Please be careful that `relativistic="full"` version can be used for SOC or non-SOC calculation, but `relativistic="scalar"` version only can be used for non-SOC calculation.
```

File: docs/advanced/scf/soc.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+When full-relativistic pseudopotential is used for non-SOC calculation, ABACUS will automatic transform it to scalar-relativistic version.
```
Comment (caic99) [inline docs/advanced/scf/soc.md]:
```suggestion
When full-relativistic pseudopotential is used for non-SOC calculation, ABACUS will automatically transform it to scalar-relativistic version.
```

File: docs/advanced/scf/soc.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+Numerical atomic orbitals in ABACUS are unrelated with spin, same orbital file can be used for SOC and non-SOC calculation.
```
Comment (caic99) [inline docs/advanced/scf/soc.md]:
```suggestion
Numerical atomic orbitals in ABACUS are unrelated with spin, and same orbital file can be used for SOC and non-SOC calculation.
```

**Review by caic99 — CHANGES_REQUESTED** [review]

**Comment by dyzheng** [comment]
Please squash merge this PR

**Review by caic99 — APPROVED** [review]

**Comment by caic99** [comment]
> Please squash merge this PR

@dyzheng You can add requested changes to batch, and commit them at once.

**Pull request merged**